synedrion/src/cggmp21/protocols/presigning.rs

```diff
@@ -542,14 +555,17 @@ impl<P: SchemeParams, I: Debug + Clone + Ord + Serialize> FinalizableToNextRound
         let cap_delta = cap_gamma * self.context.k;
 
         let alpha_sum: Signed<_> = payloads.values().map(|p| p.alpha).sum();
-        let beta_sum: Signed<_> = artifacts.values().map(|p| p.beta).sum();
+        let beta_sum: Signed<_> = artifacts.values().map(|p| p.beta.expose_secret()).sum();
```

Comment:
I wonder how far should we go to protect the secrets? In this sum, for example, the safest way to do it is to actually write out an explicit loop mutating the sum inside a `SecretBox`.

Reply:
Yeah I have the same concern and a poor intuition for what "good enough" means here.

By explicit loop, do you mean something like this?

```rust
       let beta_sum =
            artifacts
                .values()
                .fold(SecretBox::new(Box::new(Signed::default())), |mut sum, p| {
                    *sum.expose_secret_mut() = sum.expose_secret().add(p.beta.expose_secret());
                    sum
                });

```

In this case here I'm not so sure it's worth it. One line below we'd `expose_secret` again to add `beta_sum` to `delta`. But perhaps `delta` is a secret too? 😱